File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.js
function solution(n, words) {
    var answer = [];
    let obj = {};
    let idx=0;

    obj[words[0]] = 1;
    for(let i =1; i<words.length; i++){
        obj[words[i]] = obj[words[i]] ? obj[words[i]]+1 : 1; 
        if(words[i-1].slice(-1) != words[i][0]){
            idx = i;
            break;
        }else if(obj[words[i]]==2){
            idx = i;
            break;
        }
    }
    
    answer = [(idx%n)+1,Math.floor(idx/n +1) ]

    return idx==0 ? [0,0] : answer;
}

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    answer = [0, 0]
    obj = {}
    arr = []

    for i in range(1, n + 1):
        obj[i] = []

    for i, word in enumerate(words):
        seq = i % n + 1

        # 이미 나온 단어
        if word in arr:
            return [seq, len(obj[seq]) + 1]

        # 끝말잇기 실패
        if i > 0 and words[i - 1][-1] != word[0]:
            return [seq, len(obj[seq]) + 1]

        obj[seq].append(word)
        arr.append(word)

    return answer